src/features/attributeSelect.tsx

```diff
@@ -59,17 +116,48 @@ const AttributeButtonPopover = <T extends ReactText>({
     return String(item).toLowerCase().includes(query.toLowerCase());
   };
   const [sliderValue, setSliderValue] = useState(0);
+
   useEffect(() => {
     setSliderValue(Number(currentValue));
   }, [isOpen, currentValue]);
 
   const formatDisplayText = (text: string): string => {
-    // TODO: for doantrang982/eng-77-decouple-display-from-output: Create formatDisplayText from configPage
-    // const match = text.match(/\[\[(.*?)\]\]/);
-    // if (match && match[1]) {
-    //   return match[1];
-    // }
-    return text;
+    try {
+      const configUid = getPageUidByPageTitle(CONFIG);
+      const attributesNode = getSubTree({
+        key: "attributes",
+        parentUid: configUid,
+      });
+      const attributeUid = getSubTree({
+        key: attributeName,
+        parentUid: attributesNode.uid,
+      }).uid;
+      
+      const template = getSettingValueFromTree({
+        key: "template",
+        parentUid: attributeUid,
+      }) || "No styling";
+      
+      const customPattern = getSettingValueFromTree({
+        key: "customPattern",
+        parentUid: attributeUid,
+      });
+      
+      const customReplacement = getSettingValueFromTree({
+        key: "customReplacement",
+        parentUid: attributeUid,
+      });
```

Comment:
`transformItem` is run on every single item in the list, so these 5 queries will be run for every item.

Let's grab this information once and use `applyFormatting` for `transformItem`.